library/common/types/c_types.h

```diff
@@ -195,6 +195,14 @@ typedef void (*envoy_on_error_f)(envoy_error error, void* context);
  */
 typedef void (*envoy_on_complete_f)(void* context);
 
+/**
+ * Called when the async HTTP stream has been cancelled by the application (FIXME: before merge.
+ * word choice).
+ * @param context, contains the necessary state to carry out platform-specific dispatch and
+ * execution.
+ */
+typedef void (*envoy_on_cancel_f)(void* context);
```

Comment:
Hmm. Do we want this?

Reply:
If we push down the cancellation state down to the core layer we need to have a bridging callback for cancel, right?

Reply:
from offline: yes we do want this as we surface onCancel to the platform.